### ISSUE TITLE
Problems pane isn't shown by default

### ISSUE THREAD

**Issue by Timmmm** [saviorisdead/RustyCode#234]
It took me a while to realise that VSC can actually parse the output of `cargo build` and nicely show a list of errors. I think this list should be shown by default after a build.

I found for user-defined tasks there is a [`showOutput: "silent"` option](https://code.visualstudio.com/Docs/editor/tasks#_output-window-behavior) which looks like it should give the right behaviour, but unfortunately I can't work out how to make a user-defined task that runs a VS Code action (like `rust.cargo.build.debug`) rather than a command line program. And also I can't find an equivalent to `showOutput` where the commands are defined in `package.json` so I have no idea how to actually achieve this but I assume it is possible somehow.

**Comment by KalitaAlexey** [comment]
Hi @Timmmm,

Actually you cannot do this, because OutputChannel is created by a code and the code decides whether the OutputChannel should be shown.

**Comment by Timmmm** [comment]
So how would one change RustyCode to show the problems pane? I tried searching for documentation about this but didn't find anything helpful, and using the developer tools in VSC doesn't seem to expose anything from the actual VSC APIs. :-/

**Comment by KalitaAlexey** [comment]
Hi @Timmmm,

I've looked at the code.
An output channel is shown.
It means that I don't understand what you want to be shown.
Please attach screenshot with red arrow aimed to what you want to be shown.

Regards,
Alexey

**Comment by W4RH4WK** [comment]
There is a dedicated pane to show errors and warnings. It is more useful when just dealing with errors and warnings, not caring about anything else.

Ctrl + Shift + M, or View > Problems

**Comment by KalitaAlexey** [comment]
Hi @W4RH4WK,

I got it. Yes there is no API for this.

**Comment by Timmmm** [comment]
This seems to be the relevant VSC issue:

https://github.com/Microsoft/vscode/issues/11399

**Comment by KalitaAlexey** [comment]
Okay. Let's wait.

**Comment by shouya** [comment]
@Timmmm How did you get vscode to parse the output of `cargo build`? I didn't find a way to do this except parsing the result manually (https://code.visualstudio.com/Docs/editor/tasks#_defining-a-problem-matcher). Would you like to share your `task.json` or whatever that makes it work?

**Comment by KalitaAlexey** [comment]
Hi @shouya,

How about https://code.visualstudio.com/docs/extensionAPI/vscode-api#_a-namediagnosticaspan-classcodeitem-id866diagnosticspan ?

**Comment by shouya** [comment]
@KalitaAlexey This is cool and I see RustyCode is using that. However I didn't get it work for my rust project. To be simple, when I run "> Cargo: build debug", the output pane shows only the output from `carog build` and nothing appears in the problem pane.

**Comment by shouya** [comment]
```json
// ${workspaceRoot}/.vscode/task.json
{
    "version": "0.1.0",
    "command": "cargo",
    "isShellCommand": false,
    "showOutput": "silent",
    "echoCommand": true,
    "tasks": [{
        "taskName": "build",
        "args": [],
        "isBuildCommand": true,
        "problemMatcher": {
            "owner": "rust",
            "fileLocation": ["relative", "${workspaceRoot}"],
            "pattern": [{
                "regexp": "(error|warning)(\\[E\\d+\\])?: (.*)",
                "severity": 1,
                "code": 2,
                "message": 3
            }, {
                "regexp": "--> (.*):(\\d+):(\\d+)",
                "file": 1,
                "line": 2,
                "column": 3,
                "message": 0
            }]
        }
    }]
}
```

So I did the parsing manually by making a task. This task works fine but I still wonder if RustyCode can take this job.

**Comment by KalitaAlexey** [comment]
It actually can. I will fix it later.

**Comment by Timmmm** [comment]
I didn't use a task. RustyCode already works with the problems pane (more
or less; there are some formatting issues) but you have to set the "use new
output format" option in the settings to true (I can't rent the exact name)
and restart vscode.

**Comment by marmistrz** [comment]
Indeed, it already works with the problems view. Unfortunately, when I compile the project, the output pane appears, effectively hiding the problems view.

Is it possible to somehow disable switching to the output view?

**Comment by KalitaAlexey** [comment]
@marmistrz,
Yes it is.
There is the "showOutput" configuration parameter.
Set is to false.

**Comment by marmistrz** [comment]
@KalitaAlexey where is that? I searched settings.json but I can't find it. Neither in Cargo.toml or Cargo.lock

**Comment by KalitaAlexey** [comment]
@marmistrz,
Are you using RustyCode?
There is no **rust.showOutput** in RustyCode.
Check out https://github.com/KalitaAlexey/vscode-rust

**Comment by marmistrz** [comment]
Yes, I'm using Rusty Code by saviorisdead. I'll try your one :) Thanks!
